presto-main/src/main/java/com/facebook/presto/operator/TrackingLookupSourceSupplier.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+public interface TrackingLookupSourceSupplier
```

Comment:
Because we already have `LookupSourceSupplier` interface but it serves different purpose: it's used between `PagesIndex` and `HashBuilderOperator` as `Supplier<LookupSource> + some additional meta info`. Those two use cases (existing `LookupSourceSupplier` and my new `TrackingLookupSourceSupplier`) are similar but different enough to have them separate.